ci-scripts/ci-runner.bash

```diff
@@ -0,0 +1,244 @@
+#!/bin/bash
+
+##############################################################################
+#
+#
+#                                SCRIPT VARIABLES
+#
+#
+##############################################################################
+scriptname=`basename $0`
+CI_FOLDER=""
+CI_PUBLIC=0
+CI_TUTORIAL=0
+TERM="${TERM:-xterm}"
+PROFILE=""
+MODULEUSE=""
+
+#
+# This function prints the script usage form
+#
+
+CI_EXITCODE=0
+
+swap_files()
+{
+    if [ $# -lt 2 ]; then
+       echo "too few arguments to swap_files()" 1>&2
+       exit 1
+    fi
+
+    tmp="${1}_save"
+    cp $1 $tmp
+    cp $2 $1
+    cp $tmp $2
+    /bin/rm $tmp
+}
+
+usage()
+{
+    cat <<EOF
+Usage: $(tput setaf 1)$scriptname$(tput sgr0) $(tput setaf 3)[OPTIONS]$(tput sgr0) $(tput setaf 2)-f <regression-folder>$(tput sgr0)
+
+    $(tput setaf 3)OPTIONS:$(tput sgr0)
+
+    $(tput setaf 3)-f | --folder$(tput sgr0) $(tput setaf 1)DIR$(tput sgr0)        ci folder, e.g. reframe-ci
+    $(tput setaf 3)-i | --invocation$(tput sgr0) $(tput setaf 1)ARGS$(tput sgr0)   invocation for modified user checks. Multiple \`-i' options are multiple invocations
+    $(tput setaf 3)-l | --load-profile$(tput sgr0) $(tput setaf 1)ARGS$(tput sgr0) sources the given file before any execution of commands
+    $(tput setaf 3)-m | --module-use$(tput sgr0) $(tput setaf 1)ARGS$(tput sgr0)   executes module use of the give folder before loading the regression
+    $(tput setaf 3)-p | --public-only$(tput sgr0)       executes only the public version of the unittests
+    $(tput setaf 3)-t | --tutorial-only$(tput sgr0)     executes only the modified/new tutorial tests
+    $(tput setaf 3)-h | --help$(tput sgr0)              prints this help and exits
+
+EOF
+} # end of usage
+
+checked_exec()
+{
+    "$@"
+    if [ $? -ne 0 ]; then
+        CI_EXITCODE=1
+    fi
+}
+
+run_tutorial_checks()
+{
+    cmd="./bin/reframe --exec-policy=async  -r -t tutorial $@"
+    echo "Running tutorial checks with \`$cmd'"
+    checked_exec $cmd
+}
+
+run_user_checks()
+{
+    cmd="./bin/reframe --exec-policy=async  -r -t production $@"
+    echo "Running user checks with \`$cmd'"
+    checked_exec $cmd
+}
+
+run_serial_user_checks()
+{
+    cmd="./bin/reframe --exec-policy=serial  -r -t production-serial $@"
+    echo "Running user checks with \`$cmd'"
+    checked_exec $cmd
+}
```

Comment:
In all the above invocations of ReFrame it is good to pass also the `--save-log-files` option, so that we can later investigate CI failures.